governator-test-junit/src/main/java/com/netflix/governator/guice/test/AnnotationBasedTestInjectorManager.java

```diff
@@ -0,0 +1,223 @@
+    private final TestPropertyOverrideAnnotationReader testPropertyOverrideAnnotationReader = new TestPropertyOverrideAnnotationReader();
+    private TestCompositeConfig testCompositeConfig;
+
+    public AnnotationBasedTestInjectorManager(Class<?> classUnderTest) {
+        inspectModulesForTestClass(classUnderTest);
+        inspectMocksForTestClass(classUnderTest);
+        inspectSpiesForTargetKeys(Elements.getElements(modulesForTestClass));
+        testCompositeConfig = new TestCompositeConfig(classLevelOverrides, methodLevelOverrides);
+        overrideModules.add(new ArchaiusTestConfigOverrideModule(testCompositeConfig));
+        injector = createInjector(modulesForTestClass, overrideModules);
+    }
+
+    /**
+     * Injects dependencies into the provided test object.
+     */
+    public void prepareTestFixture(Object testFixture) {
+        injector.injectMembers(testFixture);
+    }
+
+    public void cleanupMocks() {
+        for (Object mock : mocksToReset) {
+            Mockito.reset(mock);
+        }
+    }
+
+    public void cleanupInjector() {
+        injector.close();
+    }
+
+    protected LifecycleInjector createInjector(List<Module> modules, List<Module> overrides) {
+        return InjectorBuilder.fromModules(modules).overrideWith(overrides).createInjector();
+    }
+
+    private void inspectModulesForTestClass(Class<?> testClass) {
+        final List<Class<? extends Module>> moduleClasses = new ArrayList<>();
+
+        moduleClasses.addAll(getModulesForAnnotatedClass(testClass));
+        for (Class<?> parentClass : getAllSuperClassesInReverseOrder(testClass)) {
+            moduleClasses.addAll(getModulesForAnnotatedClass(parentClass));
+        }
+        for (Class<? extends Module> moduleClass : moduleClasses) {
+            try {
+                modulesForTestClass.add(moduleClass.newInstance());
+            } catch (InstantiationException | IllegalAccessException e) {
+                try {
+                    Constructor<?> constructor = moduleClass.getDeclaredConstructors()[0];
+                    constructor.setAccessible(true);
+                    modulesForTestClass.add((Module) constructor.newInstance());
+                } catch (Exception ex) {
+                    throw new RuntimeException("Error instantiating module " + moduleClass
+                            + ". Please ensure that the module is public and has a no-arg constructor", e);
+                }
+            }
+        }
+    }
+
+    private List<Class<? extends Module>> getModulesForAnnotatedClass(Class<?> testClass) {
+        final Annotation annotation = testClass.getAnnotation(ModulesForTesting.class);
+        if (annotation != null) {
+            return Arrays.asList(((ModulesForTesting) annotation).value());
+        } else {
+            return Collections.emptyList();
+        }
+    }
+
+    private void inspectMocksForTestClass(Class<?> testClass) {
+        getMocksForAnnotatedFields(testClass);
```

Comment:
I would move this after the for loop so that processing order for all classes is from base class (Object) down to the test class.